### DISCUSSION TITLE
추출해 올 가치가 있는 글자, 데이터 구분 문제

### DISCUSSION THREAD

**Discussion by wlsl10** [MajorIn-ABO/MajorIn_FE#22] [General]
1. 추출해올 글자들
학생증에서 글자를 추출해올 때 학생증 사진마다 결과가 달라지는데
단국대학교를 못 가져오는 경우가 있는데
우리가 가져와야 되는 정보가 이름, 학교, 학과, 학번인데
![image](https://github.com/MajorIn-ABO/MajorIn_FE/assets/77524688/03833d27-a41f-452d-a14b-9f91e2ed7b70)

사실 상 모든 정보를 가져올 필요가 있나 싶어. 특히 학번은
그래도 숫자는 틀리는 경우가 아직까지는 한 번도 없어서 가져오게 해도 성능에는 문제가 없을 것 같은데 
근데 사실 상 우리가 학번이 겹치는지에 대한 알고리즘은 딱히 없는 것 같은데 의미가 있나 싶긴 해!
또 이름도 사실상 사용되지 않고, 이름은 겹치는지 유무에 쓰이지도 않아서 딱히 필요가 없는 데이터 같아.

추가적으로 
2. 데이터 구분 문제 도 위의 문제를 고려해볼 때 도움이 될 것 같은데
추출해 온 정보를 데이터베이스에 넘겨줄 때 
어떤 정보로 취급할지에 대한 알고리즘을 짜야 하는데 
우리 학교 학생증처럼 일부 학생증을 카드로 사용하는 학생증은 학번과 카드번호를 어떻게 구분할지와
이름과 일부 다른 문자를 구분하는 방법이 애매해('대학교'나 '학과'는 해당 단어가 포함되는지로 구분)
그래서 나는 학번이랑 이름은 빼도 된다고 생각해.
근데 또 문제는 대학교인데 우리 학과처럼 단과대가 학생증에 나와있는 경우 SW융합대학 이 단과대명인 경우 대학교 추출에 어려움이 있어.
이건 어떻게 하면 좋을까? 우리 학교야 위치 지정을 하면 되지만
다른 학교 학생들까지 이용하는 게 우리 프로그램의 주요 목표인데 어려움이 있을 듯 

학생증 OCR 자체에 대한 것도 사실 우리가 OCR 기술을 신분증(모두 같아서 같은 정보가 같은 위치에 이동)이나 카드 인식(숫자만 추출해옴-숫자 길이나 형태)이라 학생증에 사용하는 것 자체에 어려움이 있긴해

**Comment by JangGusWjd** [comment]
근데 마이페이지에서 개인정보를 표시해야 될 거 같아서 이름, 학교, 학과, 학번같은 기본적인 정보는 필요할 거 같아!
그리고 글 쓰면 학교, 학과, 20학번, 19학번 정도 학번이랑, 커뮤니티의 경우에는 성 정도만 표시될 까해서 일단 저정도만 불러오는 걸로 생각했어!

**Comment by cho-ja-young** [comment]
일단 우리가 학생증을 사용하는 이유 중에 하나가 사용자 정보를 기본적으로 다 받아오려는 거라 사용에 관계없이 일단은 다 받아와야 할 것 같음!! 알고리즘은 만약에 짜는거 어려우면 우리랑 같이 고민해도 되는거고!!
